internal/gossiper/strategy.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossiper
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+type TargetStrategy[T any] interface {
+	Target(ctx context.Context, txs []T) ([]GossipContainer[T], error)
+}
+
+type TargetProposerConfig struct {
+	GossipProposerDiff  int
+	GossipProposerDepth int
+}
+
+func DefaultTargetProposerConfig() TargetProposerConfig {
+	return TargetProposerConfig{
+		GossipProposerDiff:  4,
+		GossipProposerDepth: 1,
+	}
+}
+
+type TargetProposers[T any] struct {
+	Validators ValidatorSet
+	Config     TargetProposerConfig
+}
+
+func (g *TargetProposers[T]) Target(ctx context.Context, txs []T) ([]GossipContainer[T], error) {
+	// Select next set of proposers and send gossip to them
+	proposers, err := g.Validators.Proposers(
+		ctx,
+		g.Config.GossipProposerDiff,
+		g.Config.GossipProposerDepth,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("%w: unable to fetch proposers", err)
+	}
+	if proposers.Len() == 0 {
+		return nil, errors.New("no proposers to gossip to")
```

Comment:
Going to leave this as is because I don't think this is really a valid state to be in (the network cannot reach consensus at this point). It is possible ofc, but it's a critical problem if we reach this state.

This error also results in a warn log rather than a fatal, so I think this is fine.